src/main/java/service/impl/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package service.impl;
+
+import java.util.List;
+import model.FruitTransaction;
+import service.FruitShopService;
+import strategy.OperationStrategy;
+import strategy.OperationStrategyImpl;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private OperationStrategy operationStrategy;
+
+    public FruitShopServiceImpl() {
+        operationStrategy = new OperationStrategyImpl();
+    }
```

Comment:
The code should not be implementation dependent
```suggestion
    public FruitShopServiceImpl(OperationStrategy operationStrategy) {
        this.operationStrategy = operationStrategy;
    }
```